src/bindings/generator/src/generator.py

```diff
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import sys
+from os import mkdir, listdir, remove
+from os.path import join, exists, isfile
+from typing import List, Dict, Any
+
+from xml_parser import list_api_files, parse_api_file
+from template import model_to_data_dict, render
+from model import Interface
+
+
```

Comment:
docstring would help here for long term maintenance. What's API files here? Dbus api files? XML files? Example in the hirte source tree?

Reply:
Refactored it quite a bit and added a bit of documentation to it.